src/main/java/org/broadinstitute/consent/http/service/DatabaseMatchingServiceAPI.java

```diff
@@ -163,12 +162,15 @@ private Consent findRelatedConsent(List<Integer> dataSetIdList) {
         return consent;
     }
 
-    private List<Document> findRelatedDars(List<Integer> dataSetIds){
-        return dataAccessAPI.describeDataAccessWithDataSetIdAndRestriction(dataSetIds);
+    private List<DataAccessRequest> findRelatedDars(List<Integer> dataSetIds) {
+        return dataAccessRequestDAO.findAllDataAccessRequests().stream()
+            .filter(d -> !Collections.disjoint(dataSetIds, d.getData().getDatasetIds()))
+            .collect(Collectors.toList());
     }
 
-    private RequestMatchingObject createRequestObject(Consent consent, Document dar) throws Exception {
-        String restriction = new Gson().toJson(dar.get(DarConstants.RESTRICTION, Map.class));
-        return new RequestMatchingObject(consent.getUseRestriction(), UseRestriction.parse(restriction));
+    private RequestMatchingObject createRequestObject(Consent consent, DataAccessRequest dar) {
+        DataUse dataUse = useRestrictionConverter.parseDataUsePurpose(dar);
+        UseRestriction darUseRestriction = useRestrictionConverter.parseUseRestriction(dataUse);
+        return new RequestMatchingObject(consent.getUseRestriction(), darUseRestriction);
     }
```

Comment:
This is the meat of the PR. Previously, we were storing the `UseRestriction` on the data access request as a custom field. We stopped capturing that field when we migrated from mongo to postgres. Now, we first calculate the data use, then query Ontology for the translation to a Use Restriction. We can then generate a match object with the Use Restrictions from both the DAR and the Consent.